component/gnap/as/client_test.go

```diff
@@ -0,0 +1,549 @@
+			signer:   &mockSigner{SignatureVal: []byte("signature")},
+			tokenVal: "test Success Value",
+			grantReq: nil,
+			errMsg:   "continue: empty request",
+		},
+		{
+			name:     "error requesting gnap access with invalid signer",
+			signer:   &mockSigner{SignatureErr: fmt.Errorf("signing error")},
+			tokenVal: "test Success Value",
+			grantReq: &gnap.ContinueRequest{InteractRef: ""},
+			errMsg:   "continue: signature error: signing error",
+		},
+		{
+			name:     "error continuing gnap access with http server returning 501 error",
+			signer:   &mockSigner{SignatureVal: []byte("signature")},
+			grantReq: &gnap.ContinueRequest{InteractRef: ""},
+			errMsg:   "continue: Auth server replied with invalid Status [/gnap/continue]: 501 Not Implemented",
+		},
+		{
+			name:     "error continuing gnap access with bad http client error",
+			signer:   &mockSigner{SignatureVal: []byte("signature")},
+			grantReq: &gnap.ContinueRequest{InteractRef: ""},
+			errMsg:   "continue: failed to post HTTP request to [/gnap/continue]:",
+		},
+		{
+			name:     "error continuing gnap access with bad response unmarshall",
+			signer:   &mockSigner{SignatureVal: []byte("signature")},
+			grantReq: &gnap.ContinueRequest{InteractRef: ""},
+			errMsg:   "continue: read response not properly formatted [/gnap/continue, unexpected end of JSON input]",
+			grantResp: &gnap.ResponseContinue{
+				URI: "mocking empty response",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		tc := tt
+
+		t.Run(tc.name, func(t *testing.T) {
+			hf := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				err := processPOSTContinueRequest(w, r, tc.grantResp)
+				require.NoError(t, err)
+			})
+
+			server, url, httpClient := CreateMockHTTPServerAndClient(t, hf)
+
+			switch tc.name {
+			case "error continuing gnap access with http server returning 501 error":
+				server, url, httpClient = CreateMockHTTPServerAndClientNotOKStatusCode(t, http.StatusNotImplemented)
+			case "error continuing gnap access with bad http client error":
+				httpClient = &http.Client{}
+			case "error continuing gnap access with invalid server URL":
+				url = "bad url"
+			}
+
+			defer func() {
+				e := server.Close()
+				require.NoError(t, e)
+			}()
+
+			c, err := NewClient(tc.signer, httpClient, url)
+			require.NoError(t, err)
+
+			response, err := c.Continue(tc.grantReq)
+			if tc.errMsg != "" {
+				if tc.errMsg[len(tc.errMsg)-1] == ':' {
+					require.Contains(t, err.Error(), tc.errMsg)
+				} else {
+					require.EqualError(t, err, tc.errMsg)
+				}
+				require.Empty(t, response)
+
+				return
+			}
+
+			require.NoError(t, err)
+			require.EqualValues(t, tc.tokenVal, response.AccessToken.Value)
+		})
+	}
+}
+
+func processPOSTAuthAccessRequest(w http.ResponseWriter, r *http.Request, expectedGnapResp *gnap.AuthResponse) error {
+	if valid := validateHTTPMethod(w, r); !valid {
+		return fmt.Errorf("http method invalid")
```

Comment:
probably we can use errors.New as we are not passing any format here - same comment for other places as well.

Reply:
done